cassandra/src/main/java/se/tre/freki/storage/cassandra/CassandraStore.java

```diff
@@ -371,19 +379,117 @@ public LabelId apply(final ResultSet rows) {
 
   @Nonnull
   @Override
-  public ListenableFuture<LabelMeta> getMeta(final LabelId uid,
-                                             final LabelType type) {
-    throw new UnsupportedOperationException("Not implemented yet");
+  public ListenableFuture<Optional<LabelMeta>> getMeta(final LabelId id,
+                                                       final LabelType type) {
+
+    final ListenableFuture<Optional<Row>> metas = getOptionalMeta(id, type);
+    return transform(metas, new Function<Optional<Row>, Optional<LabelMeta>>() {
+      @Nullable
+      @Override
+      public Optional<LabelMeta> apply(final Optional<Row> result) {
+        if (!result.isPresent()) {
+          return Optional.absent();
+        }
+
+        final Row row = result.get();
+
+        return Optional.of(
+            LabelMeta.create(id, type, row.getString("name"), row.getString("meta"), 0));
+      }
+    });
+
+  }
+
+  @Nonnull
+  ListenableFuture<Optional<Row>> getOptionalMeta(final LabelId id,
+                                                  final LabelType type) {
+
+    final ListenableFuture<List<Row>> rowFuture = checkForDuplicates(id, type,
+        getMetaStatement.bind(toLong(id), type.toValue()));
+
+    return transform(rowFuture, new FirstOrAbsentFunction<Row>());
+  }
+
+  /**
+   * Use this function to check for duplicates.
+   *
+   * @param id The LabelId we were looking for in the statement. For logging purpose.
+   * @param type The type we were looking for in the statement. For logging purpose.
+   * @param statement The statement we wish to execute.
+   * @return Returns a list of rows.
+   */
+  @Nonnull
+  ListenableFuture<List<Row>> checkForDuplicates(final LabelId id,
+                                                 final LabelType type,
+                                                 final BoundStatement statement) {
+
+    final ResultSetFuture future = session.executeAsync(statement);
+
+    return transform(future, new Function<ResultSet, List<Row>>() {
+      @Nullable
+      @Override
+      public List<Row> apply(final ResultSet result) {
+
+        final List<Row> rows = result.all();
+
+        if (rows.size() > 1) {
+          LOG.error("Found duplicate IDs for ({}, {})", id, type);
+        }
+
+        return result.all();
+      }
+    });
+  }
+
+  /**
+   * Use this function to check for duplicates.
+   *
+   * @param name The name we were looking for in the statement. For logging purpose.
+   * @param type The type we were looking for in the statement. For logging purpose.
+   * @param statement The statement we wish to execute.
+   * @return Returns a list of rows.
+   */
+  @Nonnull
+  ListenableFuture<List<Row>> checkForDuplicates(final String name,
```

Comment:
Can we rename the two methods with this name? We currently use the `check` naming convention for methods that throw an exception for bad input which these do not do. I feel that `fetchNameRow` and `fetchIdRow` would be more appropriate?